@app/server/src/middleware/installSession.ts

```diff
@@ -75,7 +75,15 @@ export default (app: Express) => {
    * different authentication method such as bearer tokens.
    */
   const wrappedSessionMiddleware: RequestHandler = (req, res, next) => {
-    if (req.isSameOrigin) {
+    const origins = [];
+    if (process.env.SESSION_ALLOWED_ORIGINS) {
+      origins.push(
+        ...(process.env.SESSION_ALLOWED_ORIGINS?.replace(/s\s/g, "").split(
+          ","
+        ) || [])
+      );
+    }
+    if (req.isSameOrigin || origins.includes(req.get("Origin") || "")) {
```

Comment:
I think we should rename `isSameOrigin` to `isAllowedOrigin` and move this logic there.

Also, please process environmental variables in the root scope, near the top of the file, rather than within the middleware itself. At the moment this replace/split has to be executed for every single request which is an unnecessary cost.